### ISSUE TITLE
postgres: Support creating partitioned tables

### ISSUE THREAD

**Issue by gitmalong** [SeaQL/sea-orm#1962]
Hi!

The [TableCreateStatement](https://docs.rs/sea-orm-migration/0.12.4/sea_orm_migration/prelude/struct.TableCreateStatement.html) does not support [creating partitioned tables](https://www.postgresql.org/docs/current/ddl-partitioning.html). 

For a full blown partition support there are some more expressions that need to be supported but I think the support for creating tables with `PARTITION BY RANGE` is already a good starting point.

```sql
CREATE TABLE measurement (
    city_id         int not null,
) PARTITION BY RANGE (logdate);
```

Reasoning:
* I see this feature request  plus #1922 as the necessary features to make SeaORM + Postgres work well with huge amounts of data.

Sponsored some beer (:

Thanks

**Comment by tyt2y3** [comment]
Thank you for the sponsorship. I am not sure where there is an issue for `PARTITION BY RANGE` on sea-query. I'll assess the difficult / effort needed.

**Comment by gitmalong** [comment]
Hi. You're welcome. I created a corresponding issue: https://github.com/SeaQL/sea-query/issues/722 BR Marius

**Comment by Expurple** [comment]
FYI: if you have a regular web app and you statically know the schema that you want, you can always just write migrations in raw SQL.

This feature is only a blocker if you dynamically generate migration statements based on some conditions. It would still be nice to support this use case, of course.

**Comment by ErdemGKSL** [comment]
Yeah we can code it on raw sql but, why dont we have support for this? I mean i love using fluent api of sea orm migrations, but when i need partitions i cant...

**Comment by ErdemGKSL** [comment]
I created a pr for that, see: SeaQL/sea-query#1039
